frontend/express/public/javascripts/countly/countly.template.js

```diff
@@ -2646,6 +2648,24 @@ var AppRouter = Backbone.Router.extend({
             this.pageScripts[view] = [];
         this.pageScripts[view].push(callback);
     },
+    addWidgetCallbacks: function(widget, options){
+        if(!this.widgetCallbacks[widget]){
+            this.widgetCallbacks[widget] = {};
+        }
+        Object.assign(this.widgetCallbacks[widget], options);
+    },
+    getWidgetCallbacks: function(){
+        return this.widgetCallbacks;
+    },
+    addReportsCallbacks: function(widget, options){
+        if(!this.reportCallbacks[widget]){
+            this.reportCallbacks[widget] = {};
+        }
+        Object.assign(this.reportCallbacks[widget], options);
+    },
+    getReportsCallbacks: function(){
+        return this.reportCallbacks;
+    },
```

Comment:
I actually thought these methods would either be added by plugin or be in plugin's view as static methods. I don't think it makes sense to add them to core. So the easiest way to fix this if plugins themselves add them to `app` object. And then other plugins that use it, firstly need to check if method exists `if(app.addReportsCallbacks)` then add callback. etc